src/m365/spo/commands/serviceprincipal/serviceprincipal-permissionrequest-list.ts

```diff
@@ -58,6 +60,32 @@ class SpoServicePrincipalPermissionRequestListCommand extends SpoCommand {
       this.handleRejectedPromise(err);
     }
   }
+
+  private async getoAuth2Permissiongrants(sPOClientExtensibilityWebApplicationPrincipalId: string): Promise<string[]> {
+    const requestOptions: CliRequestOptions = {
+      url: `https://graph.microsoft.com/beta/oAuth2Permissiongrants/?$filter=clientId eq '${sPOClientExtensibilityWebApplicationPrincipalId}' and consentType eq 'AllPrincipals'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const response: any = await request.get(requestOptions);
+    return response.value[0].scope.split(' ');
+  }
+
+  private async getSPOClientExtensibilityWebApplicationPrincipalId(): Promise<string> {
+    const requestOptions: CliRequestOptions = {
+      url: `https://graph.microsoft.com/v1.0/servicePrincipals/?$filter=displayName eq 'SharePoint Online Client Extensibility Web Application Principal'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const response: any = await request.get(requestOptions);
```

Comment:
Since we're referencing properties from this object, let's not use `any` and instead define the shape of the returned object. Since this is a Graph request, we can get it from Graph typings which we already have installed.